src/sqlfluff/rules/L052.py

```diff
@@ -152,10 +152,11 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
                             break
                         except ValueError:  # pragma: no cover
                             pass
-            else:
-                self.logger.error(
+            elif memory != dict(ended_statements=deque()):
+                # Unless it's an empty statement we shouldn't reach here
+                raise ValueError(
                     "Unable to identify statement terminated by %r", context.segment
```

Comment:
Now that we're not using logging, I don't _think_ `%r` and `,` will work correctly.
```suggestion
                    f"Unable to identify statement terminated by {context.segment!r}"
```